evident/q2/_methods.py

```diff
@@ -1,101 +1,137 @@
 from typing import List
 
 import pandas as pd
-from qiime2 import CategoricalMetadataColumn, Metadata
+from qiime2 import Metadata
 from skbio import DistanceMatrix
 
-from evident import AlphaDiversityHandler, BetaDiversityHandler
-from evident.diversity_handler import (
-    RepeatedMeasuresAlphaDiversityHandler as RDH
-)
+from evident import UnivariateDataHandler, MultivariateDataHandler
+from evident.data_handler import RepeatedMeasuresUnivariateDataHandler as RDH
 from evident.effect_size import (effect_size_by_category,
                                  pairwise_effect_size_by_category)
 
 
-def alpha_power_analysis(
-    alpha_diversity: pd.Series,
-    sample_metadata: CategoricalMetadataColumn,
+def _check_provided_univariate_data(data, sample_metadata, data_column):
+    """Check if provided univariate data is valid."""
+    if data is not None:
+        if data_column is not None:
+            raise ValueError(
+                "Cannot provide a value for data_column if data is "
+                "provided."
+            )
+    else:
+        if data_column is None:
+            raise ValueError(
+                "If not providing an input for data, a value must be "
+                "provided for data_column to use a sample metadata column "
+                "identifier."
+            )
+        if data_column not in sample_metadata.columns:
+            raise ValueError(f"{data_column} not found in sample metadata.")
+        if sample_metadata[data_column].dtype.kind not in ["i", "f"]:
```

Comment:
I think pandas' `select_dtypes` would do the trick here, check this SO answer:
https://stackoverflow.com/a/43982712/379593